up-transport-vsomeip/tests/publisher_subscriber.rs

```diff
@@ -0,0 +1,159 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use log::{info, trace};
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::time::Instant;
+use up_rust::{UListener, UMessage, UMessageBuilder, UPayloadFormat, UStatus, UTransport, UUri};
+use up_transport_vsomeip::UPTransportVsomeip;
+
+const TEST_DURATION: u64 = 1000;
+
+pub struct SubscriberListener {
+    received_publish: AtomicUsize,
+}
+impl SubscriberListener {
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        Self {
+            received_publish: AtomicUsize::new(0),
+        }
+    }
+
+    pub fn received_publish(&self) -> usize {
+        self.received_publish.load(Ordering::SeqCst)
+    }
+}
+#[async_trait::async_trait]
+impl UListener for SubscriberListener {
+    async fn on_receive(&self, msg: UMessage) {
+        trace!("{:?}", msg);
+        self.received_publish.fetch_add(1, Ordering::SeqCst);
+
+        let Some(payload_bytes) = msg.payload else {
+            panic!("No bytes included in payload");
+        };
+
+        let Ok(payload_string) = std::str::from_utf8(&payload_bytes) else {
+            panic!("Unable to convert back to payload_string");
+        };
+
+        info!("We received payload_string: {payload_string}");
+    }
+
+    async fn on_error(&self, err: UStatus) {
+        trace!("{:?}", err);
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn publisher_subscriber() {
+    env_logger::init();
+
+    let authority_name = "foo";
+
+    let ue_id = 10;
+    let subscriber_ue_id = 20;
```

Comment:
@PLeVasseur 
Hey Pete,  
what is this example testing? is it testing the launch of a client and server or are they both communicating?
If communicating I see the ue_id and subscriber_ue_id different... wanted to see how they talk to each other?
 
based on Spec what I understand is ue_id is a combination of instance id and service id https://github.com/eclipse-uprotocol/up-spec/blob/main/up-l1/someip.adoc#uuris
 
shouldn't they be same for client and service that wants to talk to each other?

Reply:
Hey @DivyaVuppala -- the subscriber_ue_id in this case is probably badly worded.

This is used within the UPTransportVsomeip in order to spin up a vsomeip application with client_id and application name of subscriber_ue_id.

I'll have to better address this confusing nature. Thanks for the heads up.